### ISSUE TITLE
请问这行代码的作用是什么

### ISSUE THREAD

**Issue by xbh301** [LiuYunlong99/RQ-LLIE#5]
![image](https://github.com/LiuYunlong99/RQ-LLIE/assets/127025791/2bf50ae9-c1a4-4749-bb34-d6cba46f4a3f)
为什么要截断梯度？不能直接用z_q吗？也就是z_q=z_q

**Comment by LiuYunlong99** [comment]
z_q = z + (z_q - z).detach() 可以保证量化过程是可微分的，即使反向传播时梯度计算绕过z_q 连接到z上

**Comment by xbh301** [comment]
感谢您抽时间回复。还有一点不理解，这样不会绕过中间的codebook，只更新encoder和decoder吗？第一阶段的目的不是训练codebook和decoder吗？我在调试中运行完self.optimizer_G.step()后，看nn.embeding.weight的权重并没有变，感觉codebook没有训练。当然我在尝试直接用z_q = z_q，权重也不会改变。我对此感到很疑惑，似乎codebook并没有像预期一样训练，还是说nn.embeding.weight的参数无法训练？

**Comment by LiuYunlong99** [comment]
Codebook是通过公式(2)中的L_vq进行更新的，且仅在stage1中进行更新

**Comment by xbh301** [comment]
![image](https://github.com/LiuYunlong99/RQ-LLIE/assets/127025791/9a20f9ba-4409-44db-9e7f-7bf82a80f135)
请问第二阶段计算l_final = 0 * loss_alpha + loss_bottle中为什么要乘0，这样的话loss_alpha有什么用？
![image](https://github.com/LiuYunlong99/RQ-LLIE/assets/127025791/81967be6-fb86-41db-b8a6-53a2411d5763)
这个损失函数不就没了

**Comment by LiuYunlong99** [comment]
我们上传的代码出了问题，现在已经更改，感谢您的反馈！
